### PR TITLE
Next Step Script Changes

### PR DESCRIPTION
**Pull request by EGCGDanForth**
New next step script for spectramax step.
Modified next step script for sample receipt step due to change of UDF

### CODE REVIEW

File: scripts/next_step_assignment_spectramax.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        for next_action in next_actions:
+
+            if self.process.outputs_per_input(next_action['artifact'].id,ResultFile=True)[0].udf.get('Repeat Picogreen with 1:10 Dilution?'):
+                # update the next action to the third step from this step i.e. Picogreen 1-10 Dilution
+                next_step_object = steps[steps.index(current_step) + 3]
+                next_action['action'] = 'nextstep'
+                next_action['step'] = next_step_object
+
+            elif next_action['artifact'].samples[0].udf.get('Prep Workflow') == 'KAPA DNA Sample Prep'\
+                    or next_action['artifact'].samples[0].udf.get('PreSeqLab Fragment Analyser Complete'):
```
[the remark this answers is not in the document]
Comment (tcezard) [reply scripts/next_step_assignment_spectramax.py]: Withdraw the comment. The UDF is a boolean. However the type of UDFs is not very consistent across the LIMS.

File: scripts/next_step_assignment_spectramax.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    _use_load_config = False  # prevent the loading of the config file
+    """
+    This script assigns the next step to samples in the Spectramax Picogreen step at initial sample QC. 
+    Samples with output udf 'Repeat Picogreen with 1:10 Dilution?' as true will be directed to a dilution step.
+    Samples that have the sample udf 'Prep Workflow' as KAPA DNA Sample Prep will skip Fragment Analyser
+    and be queued to QC Review. 
+    Samples that have gone through the 1:10 Dilution and already completed the Fragment Analyser step will be queued 
+    to the QC Review step.
+    All other samples with will be queued to Fragment Analyser. 
+    """
```
[the remark this answers is not in the document]
Comment (EGCGDanForth) [reply scripts/next_step_assignment_spectramax.py]: updated

File: scripts/next_step_assignment_spectramax.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                next_action['step'] = next_step_object
+
+            elif next_action['artifact'].samples[0].udf.get('Prep Workflow') == 'KAPA DNA Sample Prep'\
+                    or next_action['artifact'].samples[0].udf.get('PreSeqLab Fragment Analyser Complete'):
+                #update the next action to the second step from this step i.e. QC Review
+                next_step_object = steps[steps.index(current_step) + 2]
+                next_action['action'] = 'nextstep'
+                next_action['step'] = next_step_object
+
+            # all other artifacts should be route to remove from workflow as the submitted sample will reassigned elsewhere
```
[the remark this answers is not in the document]
Comment (EGCGDanForth) [reply scripts/next_step_assignment_spectramax.py]: comment updated

File: scripts/next_step_assignment_spectramax.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        current_step = self.process.step.configuration  # configuration gives the ProtocolStep entity.
+        protocol = Protocol(self.process.lims, uri='/'.join(self.process.step.configuration.uri.split('/')[:-2]))
+
+        steps = protocol.steps  # a list of all the ProtocolSteps in protocol
+
+        for next_action in next_actions:
+
+            if self.process.outputs_per_input(next_action['artifact'].id,ResultFile=True)[0].udf.get('Repeat Picogreen with 1:10 Dilution?'):
+                # update the next action to the third step from this step i.e. Picogreen 1-10 Dilution
+                next_step_object = steps[steps.index(current_step) + 3]
```
Comment (mwhamgenomics) [inline scripts/next_step_assignment_spectramax.py]: We're relying here on the step order - if this changes and we forget to update this, we could get incorrect next step assignments without knowing. Is there any way of checking the step name?

Comment (EGCGDanForth) [reply scripts/next_step_assignment_spectramax.py]: Updated to use the config file to specify the step object rather than using numbers in a list.

**Pull request merged**